src/Containers/PatternListContainer.jsx

```diff
@@ -69,6 +86,35 @@ class PatternListContainer extends React.Component<Props, State> {
         await moiraApi.delPattern(pattern);
         this.getData(this.props);
     }
+
+    sortPatterns(patterns: Array<Pattern>): Array<Pattern> {
+        const { sortingColumn, sortingDown } = this.state;
+        const sorting = {
+            trigger: (x, y) => {
+                const valA = x.triggers.length || 0;
+                const valB = y.triggers.length || 0;
+                if (valA < valB) {
+                    return sortingDown ? -1 : 1;
+                }
+                if (valA > valB) {
+                    return sortingDown ? 1 : -1;
+                }
+                return 0;
+            },
+            metric: (x, y) => {
+                const valA = x.metrics.length || 0;
+                const valB = y.metrics.length || 0;
+                if (valA < valB) {
+                    return sortingDown ? -1 : 1;
+                }
+                if (valA > valB) {
+                    return sortingDown ? 1 : -1;
+                }
+                return 0;
+            },
+        };
+        return patterns.sort(sorting[sortingColumn]);
```

Comment:
Let's do `slice` before `sort`, because `sort` modify source array.

Reply:
Yeah you right :+1: